app/src/main/java/com/duckduckgo/app/onboarding/store/OnboardingStoreImpl.kt

```diff
@@ -0,0 +1,181 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.onboarding.store
+
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.core.content.edit
+import com.duckduckgo.app.browser.R
+import com.duckduckgo.app.cta.ui.DaxBubbleCta.DaxDialogIntroOption
+import com.duckduckgo.app.onboarding.ui.page.extendedonboarding.OnboardingExperimentPixel.PixelName.ONBOARDING_SEARCH_MIGHTY_DUCK
+import com.duckduckgo.app.onboarding.ui.page.extendedonboarding.OnboardingExperimentPixel.PixelName.ONBOARDING_SEARCH_SAY_DUCK
+import com.duckduckgo.app.onboarding.ui.page.extendedonboarding.OnboardingExperimentPixel.PixelName.ONBOARDING_SEARCH_SURPRISE_ME
+import com.duckduckgo.app.onboarding.ui.page.extendedonboarding.OnboardingExperimentPixel.PixelName.ONBOARDING_SEARCH_WEATHER
+import com.duckduckgo.app.onboarding.ui.page.extendedonboarding.OnboardingExperimentPixel.PixelName.ONBOARDING_VISIT_SITE_EBAY
+import com.duckduckgo.app.onboarding.ui.page.extendedonboarding.OnboardingExperimentPixel.PixelName.ONBOARDING_VISIT_SITE_ESPN
+import com.duckduckgo.app.onboarding.ui.page.extendedonboarding.OnboardingExperimentPixel.PixelName.ONBOARDING_VISIT_SITE_SURPRISE_ME
+import com.duckduckgo.app.onboarding.ui.page.extendedonboarding.OnboardingExperimentPixel.PixelName.ONBOARDING_VISIT_SITE_YAHOO
+import com.duckduckgo.mobile.android.R.drawable
+import java.util.Locale
+import javax.inject.Inject
+
+class OnboardingStoreImpl @Inject constructor(private val context: Context) : OnboardingStore {
+
+    private val preferences: SharedPreferences by lazy { context.getSharedPreferences(FILENAME, Context.MODE_PRIVATE) }
+
+    override var onboardingDialogJourney: String?
+        get() = preferences.getString(ONBOARDING_JOURNEY, null)
+        set(dialogJourney) = preferences.edit { putString(ONBOARDING_JOURNEY, dialogJourney) }
+
+    override fun getSearchOptions(): List<DaxDialogIntroOption> {
+        val country = Locale.getDefault().country
+        val language = Locale.getDefault().language
+
+        return listOf(
+            DaxDialogIntroOption(
+                optionText = if (language == "en") {
+                    context.getString(R.string.onboardingSearchDaxDialogOption1English)
+                } else {
+                    context.getString(R.string.onboardingSearchDaxDialogOption1)
+                },
+                iconRes = drawable.ic_find_search_16,
+                link = if (language == "en") "how to say duck in spanish" else context.getString(R.string.onboardingSearchQueryOption1),
+                pixel = ONBOARDING_SEARCH_SAY_DUCK,
+            ),
+            DaxDialogIntroOption(
+                optionText = if (country == "US") "mighty ducks cast" else context.getString(R.string.onboardingSearchDaxDialogOption2),
```

Comment:
Just wondering, since this condition is based on region rather than language, should we translate "mighty ducks cast" to other languages? For example, if I set my device locale to `es_US`, then this particular suggestion remains in English, while others are in Spanish.

Reply:
Good point! The O-DRI also found that inconsistency so I'm creating a new Smarling job to translate the string and release it in a week. Meanwhile we wanted to include the rest in today's release.